boards/arm/rddrone_fmuk66/board.c

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2022, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/init.h>
+#include <zephyr/device.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+
+LOG_MODULE_REGISTER(rddrone_fmuk66_board_init);
+
+/* Initialization chain of rddrone fmuk66 board requires some delays before on board sensors
+ * could be accessed after power up. In particular bme680 and bmm150 sensors require,
+ * respectively 2ms and 1ms power on delay. In order to avoid delays sum, common delay is
+ * introduced in the board start up file. Below asserts ensure correct initialization order:
+ * on board regulators, board init (this), sensors init.
+ */
+#if defined(CONFIG_REGULATOR_FIXED)
+BUILD_ASSERT(CONFIG_RDDRONE_FMUK66_INIT_PRIORITY > CONFIG_REGULATOR_FIXED_INIT_PRIORITY,
+	"CONFIG_RDDRONE_FMUK66_INIT_PRIORITY must be higher than CONFIG_REGULATOR_FIXED_INIT_PRIORITY");
+#endif
+
+#if defined(CONFIG_SENSOR)
+BUILD_ASSERT(CONFIG_RDDRONE_FMUK66_INIT_PRIORITY < CONFIG_SENSOR_INIT_PRIORITY,
+	"CONFIG_RDDRONE_FMUK66_INIT_PRIORITY must be less than CONFIG_SENSOR_INIT_PRIORITY");
+#endif
+
+
+
+static int setup()
+{
+
+	if (IS_ENABLED(CONFIG_SENSOR)) {
```

Comment:
Are these delays specific to the RDDRONE? If not, I would rather see them integrated into the sensor driver. This could be done with some sort of KConfig, which enabled the sensor driver to delay at startup so the sensor could power up.

Reply:
These changes are specific to rddrone fmuk66. 
There is a 3.3V supply for several onboard sensor, which is enabled through a GPIO pin.
Changes in this file make sure that the sequence of enablement of GPIO module, regulator driver, this board specific module and the sensor modules are in a particular sequence and have a small delay between their respective enablement

Reply:
Thanks for the response, I'm fine leaving this delay where it currently is in that case.